src/GraphQL/DataLoader/SearchLocationLoader.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformGraphQL\GraphQL\DataLoader;
+
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use EzSystems\EzPlatformGraphQL\GraphQL\DataLoader\Exception\ArgumentsException;
+use eZ\Publish\API\Repository\Exceptions as ApiException;
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Search\SearchHit;
+
+/**
+ * @internal
+ */
+class SearchLocationLoader implements LocationLoader
+{
+    /**
+     * @var SearchService
+     */
+    private $searchService;
+
+    /**
+     * @var LocationService
+     */
+    private $locationService;
+
+    public function __construct(SearchService $searchService, LocationService $locationService)
+    {
+        $this->searchService = $searchService;
+        $this->locationService = $locationService;
+    }
+
+    public function find(LocationQuery $query): array
+    {
+        return array_map(
+            function (SearchHit $searchHit) {
+                return $searchHit->valueObject;
+            },
+            $this->searchService->findLocations($query)->searchHits
+        );
+    }
+
+    public function findById($id): Location
+    {
+        try {
+            return $this->locationService->loadLocation($id);
+        } catch (ApiException\InvalidArgumentException $e) {
+        } catch (ApiException\NotFoundException $e) {
+            throw new ArgumentsException($e->getMessage(), $e->getCode(), $e);
```

Comment:
There happens something confusing to me. Why does `ArgumentsException` extend `NotFoundException` if it thrown for `NotFoundException` and for `InvalidArgumentException` below? To my understanding it should not extend any particular Exception class because it's very general exception. Also do we need specific `ArgumentsException` class if we have `InvalidArgumentException` from our API?

Reply:
The general idea is to make that package only throw its own exceptions (it supposedly catches everything else). Why it extends `NotFoundException`... I don't know. A mistake I guess. I'll change it extend `\Exception`. Would that work for you ?

Generally speaking, error handling in GraphQL needs to be vastly improved, but I have barely scratched the surface of it.

Reply:
After thinking about it, I'd like to request your approval to leave this aside for now. The reason why it is implemented that way in this loader is that it was implemented that way in the existing ones.

We must find a proper way to implement exception handling in loaders, but I don't think that we must do it in the context of that PR.